qiita_db/environment_manager.py

```diff
@@ -127,6 +128,36 @@ def _download_reference_files():
         _insert_processed_params(ref)
 
 
+def create_mountpoints():
+    r"""In a fresh qiita setup, sub-directories under
+        qiita_config.base_data_dir might not yet exist. To avoid failing in
+        later steps, they are created here.
+    """
+    with qdb.sql_connection.TRN:
+        # Insert the settings values to the database
+        sql = """SELECT mountpoint FROM qiita.data_directory
+                 WHERE active = TRUE"""
+        qdb.sql_connection.TRN.add(sql)
+        created_subdirs = []
+        for subdir in qdb.sql_connection.TRN.execute_fetchflatten():
```

Comment:
If I remember correctly, it has been a while since I have worked with the qiita mounts, concatenating the "test_data") is actually given by the subdirectory flag, I think [this code illustrates that](https://github.com/qiita-spots/qiita/blob/master/qiita_db/util.py#L566-L582). Maybe worth relying on that method to get the correct expected folder. An idea, could be to change the query in 130 to select all the unique mount points and then use get_mountpoint to get the expected folder and create it. What do you think?

Reply:
A very good idea. I changed accordingly